modules/drivers/druid-jdbc/src/metabase/driver/druid_jdbc.clj

```diff
@@ -0,0 +1,183 @@
+(ns metabase.driver.druid-jdbc
+  (:require
+   [cheshire.core :as json]
+   [clj-http.client :as http]
+   [clojure.string :as str]
+   [clojure.walk :as walk]
+   [java-time.api :as t]
+   [metabase.driver :as driver]
+   [metabase.driver.common :as driver.common]
+   [metabase.driver.sql-jdbc.connection :as sql-jdbc.conn]
+   [metabase.driver.sql-jdbc.execute :as sql-jdbc.execute]
+   [metabase.driver.sql-jdbc.sync :as sql-jdbc.sync]
+   [metabase.driver.sql.query-processor :as sql.qp]
+   [metabase.driver.sql.query-processor.util :as sql.qp.u]
+   [metabase.driver.sql.util.unprepare :as unprepare]
+   [metabase.lib.field :as lib.field]
+   [metabase.lib.metadata :as lib.metadata]
+   [metabase.query-processor.store :as qp.store]
+   [metabase.query-processor.util.add-alias-info :as add]
+   [metabase.util.honey-sql-2 :as h2x]
+   [metabase.util.log :as log])
+  (:import
+   (java.sql ResultSet Types)
+   (java.time ZonedDateTime)))
+
+(set! *warn-on-reflection* true)
+
+(driver/register! :druid-jdbc :parent :sql-jdbc)
+
+(doseq [[feature supported?] {:set-timezone            true
+                              :expression-aggregations true}]
+  (defmethod driver/database-supports? [:druid-jdbc feature] [_driver _feature _db] supported?))
+
+(defmethod sql-jdbc.conn/connection-details->spec :druid-jdbc
+  [_driver {:keys [host port] :as _db-details}]
+  (merge {:classname   "org.apache.calcite.avatica.remote.Driver"
+          :subprotocol "avatica:remote"
+          :subname     (str "url=" host ":" port "/druid/v2/sql/avatica/;transparent_reconnection=true")}
+         (when (some? (driver/report-timezone))
+           {:sqlTimeZone (driver/report-timezone)})))
+
+(defmethod driver/db-default-timezone :druid-jdbc
+  [_driver _database]
+  "UTC")
+
+(defmethod driver/db-start-of-week :druid-jdbc
+  [_]
+  :monday)
+
+(defmethod sql-jdbc.sync/database-type->base-type :druid-jdbc
+  [_ database-type]
+  ({:TIMESTAMP            :type/DateTime
+    :VARCHAR              :type/Text
+    :DECIMAL              :type/Decimal
+    :FLOAT                :type/Float
+    :REAL                 :type/Float
+    :DOUBLE               :type/Float
+    :BIGINT               :type/BigInteger
+    :COMPLEX<json>        :type/JSON
+    :COMPLEX<hyperUnique> :type/DruidHyperUnique}
+   database-type
+   :type/*))
+
+(defmethod sql-jdbc.execute/read-column-thunk [:druid-jdbc Types/TIMESTAMP]
+  [_ ^ResultSet rs _ ^long i]
```

Comment:
```suggestion
  [_driver ^ResultSet rs _rsmeta ^Long i]
```

1. Always good to do `_rsmeta` instead of `_`
2. I'm guessing you copied `^long` (unboxed/raw 64-bit integer) from the stuff in `metabase.driver.sql-jdbc.execute`, but that stuff is wrong actually. This should 100% be getting passed as a boxed `java.lang.Long` object. So `^Long` instead of `^long`. I guess the Clojure compiler somehow emits code that does the right thing either way